lua/lspconfig/leanls.lua

```diff
@@ -6,12 +6,29 @@ configs.leanls = {
     cmd = { 'lean', '--server' },
     filetypes = { 'lean' },
     root_dir = function(fname)
-      return util.root_pattern 'leanpkg.toml'(fname) or util.find_git_ancestor(fname) or util.path.dirname(fname)
+      -- check if inside elan stdlib
+      local stdlib_dir
+      do
+        local _, endpos = fname:find(util.path.sep .. util.path.join('lib', 'lean'))
+        if endpos then
+          stdlib_dir = fname:sub(1, endpos)
+        end
+      end
+
+      -- check if inside checkout stdlib
+      local checkout_stdlib_dir
+      do
+        local _, endpos = fname:find(util.path.sep .. 'src')
+        if endpos then
+          checkout_stdlib_dir = fname:sub(1, endpos)
+        end
+      end
```

Comment:
I think this might have too many false positives, `src` directories are pretty common.  It's no big deal if we start too many lean 4 server processes, so maybe we should just remove this check.

Reply:
Done.

Reply:
Also note that in this case it should fallback to the root directory of the checkout as a git ancestor.